File: mario.c
#include <stdio.h>
#include <cs50.h>

int main (void)
{
    int height;
    do {
        printf("Height: ");
        height = get_int();
    } while (height < 0 || height > 23);

    for (int i = 1; i <= height; i++)
    {
        for (int j = height - i; j > 0; j--)
        {
            printf(" ");
        }
        for (int k = 2; k <= i + 2; k++)
        {
            printf("#");
            if (k == i + 2)
            {
                printf("\n");
            }
        }
    }
}

File: mario.py
import cs50

def main():
    # prompt user for height of half pyramid
    while True:
        print("Height: ", end="")
        height = cs50.get_int()

        # break out of loop if valid height entered
        if (height > 0 and height < 24):
            break

    # iterate over height of half pyramid
    for i in range(1, height + 1, 1):

        # print appropriate number of spaces for each row
        for j in range(height - i, 0, -1):
            print(" ", end="")

        # print appropriate number of pound chars for each row
        for k in range(2, i + 3, 1):
            print("#", end="")

            # go to next line when all spaces and pound chars have printed
            if k == i + 2:
                print()

if __name__ == "__main__":
    main()